tor/cmd_onion_test.go

```diff
@@ -184,20 +203,30 @@ type mockStore struct {
 // interface.
 var _ OnionStore = (*mockStore)(nil)
 
-func (m *mockStore) StorePrivateKey(ot OnionType, key []byte) error {
-	args := m.Called(ot, key)
+func (m *mockStore) StorePrivateKey(key []byte) error {
+	args := m.Called(key)
 	return args.Error(0)
 }
 
-func (m *mockStore) PrivateKey(ot OnionType) ([]byte, error) {
-	args := m.Called(ot)
-	if args.Get(0) == nil {
-		return nil, args.Error(1)
-	}
-	return args.Get(0).([]byte), args.Error(1)
+func (m *mockStore) PrivateKey() ([]byte, error) {
+	args := m.Called()
+	return []byte("hide_me_plz"), args.Error(1)
 }
 
-func (m *mockStore) DeletePrivateKey(ot OnionType) error {
-	args := m.Called(ot)
+func (m *mockStore) DeletePrivateKey() error {
+	args := m.Called()
 	return args.Error(0)
 }
+
+type MockEncrypter struct{}
+
+func (m MockEncrypter) EncryptPayloadToWriter(_ bytes.Buffer,
+	_ io.Writer) error {
+
+	return nil
+}
+
+func (m MockEncrypter) DecryptPayloadFromReader(_ io.Reader) ([]byte, error) {
+
+	return privateKey, nil
```

Comment:
Should we return another string here so we can test that the encrypter was actually called?

Reply:
Another string?

Reply:
Something like `this_is_the_encrypted_key` instead of the normal private key. Then in the test make sure the "encrypted" key is returned instead of the plaintext `privateKey`.

Reply:
Ahhh I see. That makes sense. I feel iffy about calling it an encrypted key since it's being returned by the decrypter, so I made one called "anotherKey." Otherwise, does the new change look like what you had in mind?